api/python/t4/packages.py

```diff
@@ -198,13 +198,13 @@ def deserialize(self, func=None, **format_opts):
 
         pkey_ext = pathlib.PurePosixPath(unquote(urlparse(physical_key).path)).suffix
 
-        # Verify format can be handled before checking hash..
-        FormatRegistry.deserialize(data, self.meta, pkey_ext, check_only=True, **format_opts)
+        # Verify format can be handled before checking hash.  Raises if none found.
+        formats = FormatRegistry.search(None, self.meta, pkey_ext)
```

Comment:
So `search` always either returns a nonempty list, or throws?

Reply:
Yes.  This was mainly to also return a reason for the failure, if there was one.